README.md

```diff
@@ -1,42 +1,44 @@
 # rubydeps
 
+The DependencyWorker currently performs the following actions:
+
+* Fetch gemfile and gemspec for a given Gem
+  * Currently, the GithubGem class specifies the needed information for a gem hosted on github. Other providers could be added.
+* Parse both files for dependency-requirements
+* Determine the latest version for each required gem
+* Generate a Hash of results of the following form:
+```ruby
+{
+  "gemspec" => [
```

Comment:
Clustering outdated gems in the results hash would be helpful when counting them. So instead of an array with the dependencies you could have another nested hash with keys `:outdated` and `:uptodate`?

Reply:
Yes, makes sense! I was also wondering whether we should make a difference between gems that are out of date only by a 'bump', and those that are actually a minor or major version behind. Do we need that for calculating the badges?

Reply:
I hadn't considered the orange badge to signal dependencies _getting out of date_ so to speak. We could add such a badge, and in that case we would need to know the extent to which dependencies are out of date (bump, minor, or major).

Reply:
I've added calculation of the difference between bump, minor, and major, but it doesn't appear to be cheap: runtime for `test.rb` (with four repos) went from <2 sec to ~9 or ~10 secs.

The problem is that it's not trivial to compare a version *requirement*, such as `> 4.1, < 5.3` to the latest *version* (such as `6.1`). That requires knowing the *highest version* that satisfies the constraint, and comparing that to the latest version. I'm currently letting RubyGems figure out what the highest version that satisfies a certain requirement is, but that means doing an extra rubygems call for (almost) every out of date dependency. 

Do you think the performance hit is worth it? Or can you think of a good way of calculating the highest version that still satisfies a certain constraint?